### ISSUE TITLE
Add documentation on `observable` decorator

### ISSUE THREAD

**Issue by PWKad** [aurelia/binding#587]
You can assign this one to me.  Need to add docs describing behaviors and suggested usage.

**Comment by Gheoan** [comment]
There is some documentation about that [here](http://aurelia.io/hub.html#/doc/article/aurelia/framework/latest/cheat-sheet/9).

**Comment by obust** [comment]
Why is it in the "Custom Element" section specifically ? Shouldn't it have its own cheatsheet section (or maybe a top level section called "Observation Patterns").
Observation is a big part of managing the application state and I think Aurelia makes it easy to  (and thus should be more advertised).

A few things that I did not realise right away but turned out helpful:
- the `@bindable` property decorator make the property observable by default (no need for `@observable`)
- the change handler is triggered when the *value* of the property changes. This means that for arrays, a `.push()` will not trigger the change handler but a reassignment to a new array with the extra value will trigger it. More generally, when dealing with observable properties, adopting a *pure functional approach* (one of redux's 3 principles "Changes are made with pure functions")  is great for triggering dynamic observations.

```js
// add value to array
this.myArray = [...this.myArray, newValue];
// remove element from array
this.myArray = this.myArray.filter(element => element !== myElementToRemove)
```
When I started doing this, it made my code much cleaner and helped propagate changes between components. I think it's a good tip to give (the docs doesn't have to teach pure functions but at least mention them or show an example).
This can help user get pretty far in managing application state without using Redux or the aurelia `BindingEngine` (which adds boilerplate code with subscription/unsubscription).

Anyway, at least I agree the docs should be more extensive on property observation :+1:
